sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -120,14 +125,20 @@ class RapidsGdsStore(
       other.getSpillPriority, other.spillCallback)
   }
 
-  class BatchSpiller() {
+  class BatchSpiller() extends AutoCloseable {
     private val blockSize = 4096
     private[this] val spilledBuffers = new ConcurrentHashMap[File, Set[RapidsBufferId]]
     private[this] val pendingBuffers = ArrayBuffer.empty[RapidsGdsBatchedBuffer]
     private[this] val batchWriteBuffer = CuFileBuffer.allocate(batchWriteBufferSize, true)
     private[this] var currentFile = TempSpillBufferId().getDiskPath(diskBlockManager)
     private[this] var currentOffset = 0L
 
+    override def close(): Unit = {
+      pendingBuffers.foreach(_.free())
```

Comment:
Do we need a notion of safeFree to avoid leaking buffers when an  in-between free fails?

Do we need batchWriteBuffer.close() in a finally to prevent it from being leaked if  a `free` above throws?

Reply:
Filed #5098 to track hardening the code to potential exceptions from `RapidsBuffer.free`.